src/services/auth/plugins/password/index.ts

```diff
@@ -1,20 +1,51 @@
 import { StatusCodes } from 'http-status-codes';
+import { ExtractJwt, Strategy } from 'passport-jwt';
 
+import fastifyPassport from '@fastify/passport';
 import { FastifyPluginAsync } from 'fastify';
 
-import { RecaptchaAction } from '@graasp/sdk';
+import { ActionTriggers, Context, RecaptchaAction } from '@graasp/sdk';
 
-import { PUBLIC_URL } from '../../../../utils/config';
+import { PASSWORD_RESET_JWT_SECRET, PUBLIC_URL } from '../../../../utils/config';
 import { UnauthorizedMember } from '../../../../utils/errors';
 import { buildRepositories } from '../../../../utils/repositories';
 import { getRedirectionUrl } from '../../utils';
-import { passwordLogin, updatePassword } from './schemas';
+import {
+  passwordLogin,
+  patchResetPasswordRequest,
+  postResetPasswordRequest,
+  updatePassword,
+} from './schemas';
 import { MemberPasswordService } from './service';
 
+const PASSPORT_STATEGY_ID = 'jwt-reset-password';
+
 const plugin: FastifyPluginAsync = async (fastify) => {
-  const { log, db } = fastify;
+  const { mailer, log, db } = fastify;
+
+  await fastify.register(fastifyPassport.initialize());
+  await fastify.register(fastifyPassport.secureSession());
 
-  const memberPasswordService = new MemberPasswordService(log);
+  fastifyPassport.use(
+    PASSPORT_STATEGY_ID,
+    new Strategy(
+      {
+        jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+        secretOrKey: PASSWORD_RESET_JWT_SECRET,
+        passReqToCallback: true,
+      },
+      async (req, _payload, done) => {
+        const token = ExtractJwt.fromAuthHeaderAsBearerToken()(req);
+        if (token && (await memberPasswordService.validateResetPasswordToken(token))) {
```

Comment:
Do we maybe want to perform this before checking to improve readability ? Also add some comments about what it means to return `done(..., false)` or `done(..., true)`